src/globus_sdk/services/gcs/data/endpoint.py

```diff
@@ -0,0 +1,146 @@
+    :param public: Flag indicating whether this endpoint is visible to all other Globus
+        users. If false, only users which have been granted a role on the endpoint or
+        one of its collections, or belong to a domain allowed access to any of its
+        storage gateways may view it.
+
+    :param gridftp_control_channel_port: TCP port for the Globus control channel to
+        listen on. By default, the control channel is passed through 443 with an ALPN
+        header containing the value "ftp".
+    :param max_concurrency: Admin-specified value when the ``network_use`` property’s
+        value is ``"custom"``; otherwise the preset value for the specified
+        ``network_use``.
+    :param max_parallelism: Admin-specified value when the ``network_use`` property’s
+        value is ``"custom"``; otherwise the preset value for the specified
+        ``network_use``.
+    :param preferred_concurrency: Admin-specified value when the ``network_use``
+        property’s value is ``"custom"``; otherwise the preset value for the specified
+        ``network_use``.
+    :param preferred_parallelism: Admin-specified value when the ``network_use``
+        property’s value is ``"custom"``; otherwise the preset value for the specified
+        ``network_use``.
+    """
+
+    DATATYPE_BASE = "endpoint"
+    DATATYPE_VERSION_IMPLICATIONS: dict[str, tuple[int, int, int]] = {
+        "gridftp_control_channel_port": (1, 1, 0),
+    }
+    DATATYPE_VERSION_CALLBACKS: tuple[DatatypeCallback, ...] = ()
+
+    # Note: The fields below represent the set of mutable endpoint fields in
+    #   an Endpoint#1.2.0 document.
+    #   https://docs.globus.org/globus-connect-server/v5.4/api/schemas/Endpoint_1_2_0_schema/
+    # Read-only fields (e.g. "gcs_manager_url", "endpoint_id", and
+    #   "earliest_last_access") are intentionally omitted as this data class is designed
+    #   for input construction not response parsing.
+    def __init__(
+        self,
+        *,
+        # data type
+        data_type: str | None = None,
+        # strs
+        contact_email: str | None = None,
+        contact_info: str | None = None,
+        department: str | None = None,
+        description: str | None = None,
+        display_name: str | None = None,
+        info_link: str | None = None,
+        network_use: (
+            t.Literal["normal", "minimal", "aggressive", "custom"] | None
+        ) = None,
+        organization: str | None = None,
+        subscription_id: str | None = None,
+        # str lists
+        keywords: t.Iterable[str] | None = None,
+        # bools
+        allow_udt: bool | None = None,
+        public: bool | None = None,
+        # ints
+        gridftp_control_channel_port: int | None = None,
+        max_concurrency: int | None = None,
+        max_parallelism: int | None = None,
+        preferred_concurrency: int | None = None,
+        preferred_parallelism: int | None = None,
```

Comment:
I've reached out to the GCS team to ask if any of these fields can be set explicitly to `null` in an update.
If there are fields which need to be nullable, we'll want to switch this to using `MISSING`.

Reply:
Saw in the thread there are 2 fields that do allow nulling which is unfortunate. The existing PayloadWrapper `_set_opt<type>` methods don't allow for null setting. It'll expand the scope of this work, but I'll need to refactor those all I guess if we're going to support explicit nulls.

Have I mentioned how much I love API nulls!

Reply:
I couldn't figure out a clean way to genericize this in the existing interfaces (specifically the batch `_set_opt<type>s` functions). This is because "opt" in those helper functions use "optional" to mean "omittable", not "nullable"). 

What I've done is make `PayloadWrapper` configurable so that extending classes can override the "omit this" sentinel & gone directly to "_set_value(...)" for nullable fields.

Reply:
Merged your PR.

Reply:
Okay, I'm doing a re-review now.